xbmc/windowing/osx/WinEventsOSXImpl.mm

```diff
@@ -389,6 +315,41 @@ - (XBMC_Event)keyPressEvent:(NSEvent*)nsEvent
   return newEvent;
 }
 
+- (void)SendXBMCMouseEvent:(NSEvent*)
+                   nsEvent:(XBMC_Event&)xbmcEvent
+                          :(uint8_t)mouseEventType
+                          :(std::optional<uint8_t>)buttonCode
+{
+  auto location = [self TranslateMouseLocation:nsEvent];
+  if (location.has_value())
+  {
+    NSPoint locationCoordinates = location.value();
+    xbmcEvent.type = mouseEventType;
+    if (mouseEventType != XBMC_MOUSEMOTION)
+    {
+      if (buttonCode.has_value())
+      {
+        xbmcEvent.button.button = buttonCode.value();
+      }
+      xbmcEvent.button.x = locationCoordinates.x;
+      xbmcEvent.button.y = locationCoordinates.y;
+    }
+    else
+    {
+      xbmcEvent.motion.x = locationCoordinates.x;
+      xbmcEvent.motion.y = locationCoordinates.y;
+    }
+    [self MessagePush:&xbmcEvent];
+
+    // scrollwhell need a subsquent button press with no button code
+    if (buttonCode.has_value() &&
+        (buttonCode.value() == XBMC_BUTTON_WHEELUP || buttonCode.value() == XBMC_BUTTON_WHEELDOWN))
```

Comment:
this should really stay at the calling site, return bool (or maybe `location` itself) from the helper to indicate that it succeeded